runtime/stdlib.js

```diff
@@ -73,7 +73,7 @@ function caml_call_gen(f, args) {
   var n = (f.l >= 0)?f.l:(f.l = f.length);
   var argsLen = args.length;
   var d = n - argsLen;
-  if (d == 0) {
+  if (n === 0 || d === 0) {
```

Comment:
This is not going to work since the last element of `args` is a continuation function, which function `f` is certainly not expecting.